src/App.tsx

```diff
@@ -1,41 +1,64 @@
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Post } from './react-app-env';
+import { Loader } from './components/Loader';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: React.FC = () => {
+  const [currentUser, setCurrentUser] = useState('0');
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [isLoading, setIsLoading] = useState(false);
 
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label>
+          Select a user: &nbsp;
 
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+          <select
+            className="App__user-selector"
+            value={currentUser}
+            onChange={(event) => {
+              setCurrentUser(event.target.value);
+              setIsLoading(true);
+            }}
+          >
+            <option value="0" disabled>All users</option>
+            <option value="1">Leanne Graham</option>
+            <option value="2">Ervin Howell</option>
+            <option value="3">Clementine Bauch</option>
+            <option value="4">Patricia Lebsack</option>
+            <option value="5">Chelsey Dietrich</option>
+            <option value="6">Mrs. Dennis Schulist</option>
+            <option value="7">Kurtis Weissnat</option>
+            <option value="8">Nicholas Runolfsdottir V</option>
+            <option value="9">Glenna Reichert</option>
+            <option value="10">Leanne Graham</option>
+          </select>
+        </label>
+      </header>
+      <main className="App__main">
+        <div className="App__sidebar">
+          {isLoading && <Loader />}
+          <PostsList
+            userSelectedId={currentUser}
+            selectPost={setSelectedPost}
+            post={selectedPost}
+            setIsLoading={setIsLoading}
+          />
+        </div>
 
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+        <div className="App__content">
+          <PostDetails
+            post={selectedPost}
+            setIsLoading={setIsLoading}
+          />
```

Comment:
You shouldn't show this component if there is no selected post
Show some string if there is no selected post